### PR TITLE
feat(makefile): archlinux support & minor updates

### PR DESCRIPTION
**Pull request by noelmcloughlin**
Fixes #

- Adds Archlinux Support
-  Fix package handling (for archlinux)
-  Few minor fixes and improvementt

Description:

*make install* tested on Ubuntu, CentOS, OpenSUSE 15, Archlinux

-----------------

**Checklist:**
- [ ] Added unittests and or functional tests
- [ ] Adapted documentation
- [ ] Referenced issues or internal bugtracker
- [ x] Ran integration tests successfully (trigger with "@susebot run teuthology" in a GitHub comment; see the [wiki](https://github.com/SUSE/DeepSea/wiki/Testing) for more information)

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
looks good, but could you elaborate on the removed hyphen `-`?

**Comment by noelmcloughlin** [comment]
Should the hyphen be there? I assumed it was left after git conflict.

File: Makefile [hunk 1]
```diff
@@ -23,20 +23,27 @@ USER=root
 GROUP=root
 ifeq ($(OS), centos)
 PKG_INSTALL=yum install -y
-PYTHON_DEPS=rh-python36-setuptools python36-click tox python36-configobj
+PY_VER=36
+PYTHON_DEPS=python${PY_VER}-setuptools python${PY_VER}-click tox python${PY_VER}-configobj
```
Comment (smithfarm) [inline Makefile]: This line appears to be changing `rh-python${PY_VER}-setuptools` to `python${PY_VER}-setuptools` for CentOS - have you tested that?

**Comment by smithfarm** [comment]
From the GNU Make manual: "To ignore errors in a recipe line, write a ‘-’ at the beginning of the line’s text (after the initial tab). The ‘-’ is discarded before the line is passed to the shell for execution."

https://www.gnu.org/software/make/manual/make.html#Errors-in-Recipes

File: Makefile (hunk 1 above)
Comment (noelmcloughlin) [reply Makefile]: Hi @smithfarm yes I tested on the OS in PR description: "make install tested on Ubuntu, CentOS, OpenSUSE 15, Archlinux".  I was checking why package had non-standard name and the other package name works and is more standard.

**Comment by noelmcloughlin** [comment]
Hi @smithfarm @jschmid1 Thanks for reviewing. I have updated PR per comments.  

Diff with previous looks good.
```
(base) :DeepSea nmcloughlin$ diff Makefile /tmp/Makefile.previous
45d44
< endif
989c988
< 	([ -n "${SALT_API}" ] && systemctl restart ${SALT_API}) || true
---
> 	([ -n "${SALT_API}" ] && systemctl restart salt-api) || true
```

**Comment by noelmcloughlin** [comment]
Successfully Re-tested (`make install`) on CentOS/Suse/Arch/Ubuntu - all good.

**Comment by noelmcloughlin** [comment]
@susebot run teuthology

**Pull request merged**